coffeecard/CoffeeCard.Library/Services/AccountService.cs

```diff
@@ -289,20 +289,25 @@ private async Task AnonymizeUser(User user)
             user.DateUpdated = DateTime.Now;
             user.PrivacyActivated = true;
             user.UserState = UserState.Deleted;
-            await _context.SaveChangesAsync();            
+            await _context.SaveChangesAsync();
         }
-        
-        private User VerifyTokenClaimAndUser(string token)
+
+        private async Task<User> VerifyTokenClaimAndUserAsync(string token)
         {
+            if (await _tokenService.ValidateTokenAsync(token))
+            {
+                throw new ApiException("The token is invalid!", StatusCodes.Status401Unauthorized);
+            }
+            
             var jwtToken = _tokenService.ReadToken(token);
             if (!jwtToken.Claims.Any(x => x.Type == ClaimTypes.Role && x.Value == "verification_token"))
-                throw new ApiException("The token is invalid!", 400);
-
+                throw new ApiException("The token is invalid!", StatusCodes.Status401Unauthorized);
+            
             var emailClaim = jwtToken.Claims.FirstOrDefault(x => x.Type == ClaimTypes.Email);
-            if (emailClaim == null) throw new ApiException("The token is invalid!", 400);
+            if (emailClaim == null) throw new ApiException("The token is invalid!", StatusCodes.Status401Unauthorized);
 
             var user = _context.Users.FirstOrDefault(x => x.Email == emailClaim.Value);
-            if (user == null) throw new ApiException("The token is invalid!", 400);
+            if (user == null) throw new ApiException("The token is invalid!", StatusCodes.Status401Unauthorized);
```

Comment:
Since the ApiException message is the same for all three, but different, cases, we should at least log an info or debug with the reason for failing else we would not be able to tell why the token failed